kafka-rest/src/main/java/io/confluent/kafkarest/controllers/AbstractConfigManager.java

```diff
@@ -188,6 +191,43 @@ final CompletableFuture<Void> safeAlterConfigs(
         .thenCompose(config -> alterConfigs(resourceId, commands));
   }
 
+  /**
+   * Atomically alter configs according to {@code commands}, checking if the configs exist first. If
+   * the {@code validateOnly} flag is set, the operation is only dry-ran (the configs do not get
+   * altered as a result).
+   */
+  // KREST-8518 A separate overload is provided instead of changing the pre-existing
+  // safeAlterConfigs method in order to minimize any risks related to external usage of that method
+  // (as this manager can be injected in projects inheriting from kafka-rest) and to minimize the
+  // amount of necessary changes (e.g. by avoiding the need to heavily refactor tests).
```

Comment:
I don't quite follow (even with the comment!) why the original safeAlterConfigs can't have its body deleted, and replaced with 

`return safeAlterConfigs(clusterId, resourceId, prototype, commands, false)`

The tests shouldn't need to change, they can still call the original method.